test/cli/install/bun-yarn.test.ts

```diff
@@ -0,0 +1,85 @@
+import { file, spawn } from "bun";
+import { afterAll, afterEach, beforeAll, beforeEach, expect, it } from "bun:test";
+import { bunExe, bunEnv as env } from "harness";
+import { access, mkdir, mkdtemp, readlink, realpath, rm, writeFile } from "fs/promises";
+import { join, relative } from "path";
+import { tmpdir } from "os";
+import {
+  dummyAfterAll,
+  dummyAfterEach,
+  dummyBeforeAll,
+  dummyBeforeEach,
+  dummyRegistry,
+  package_dir,
+  readdirSorted,
+  requested,
+  root_url,
+  setHandler,
+} from "./dummy.registry";
+
+beforeAll(dummyBeforeAll);
+afterAll(dummyAfterAll);
+
+let add_dir: string;
+
+beforeEach(async () => {
+  add_dir = await mkdtemp(join(await realpath(tmpdir()), "bun-yarn.test"));
+  await dummyBeforeEach();
+});
+
+afterEach(async () => {
+  await rm(add_dir, { force: true, recursive: true });
+  await dummyAfterEach();
+});
+
+async function command(...args: Array<string>) {
+  const { stdout, stderr, exited } = spawn({
+    cmd: [bunExe(), ...args],
+    cwd: package_dir,
+    stdout: null,
+    stdin: "pipe",
+    stderr: "pipe",
+    env,
+  });
+  expect(stderr).toBeDefined();
+  const err = await new Response(stderr).text();
+  expect(stdout).toBeDefined();
+  const out = await new Response(stdout).text();
+  return { err, out };
+}
+
+it("should wrap package title names that begin with true or false in quotes.", async () => {
+  for (const [pkg_name, pkg_ver] of [
+    ["true", "0.0.4"],
+    ["false", "0.0.4"],
+    ["falsetto", "0.2.5"],
+    ["true-case-path", "2.2.1"],
+  ]) {
+    const urls: string[] = [];
+    setHandler(dummyRegistry(urls, { [pkg_ver]: {} }));
+
+    await writeFile(
+      join(package_dir, "package.json"),
+      JSON.stringify({
+        name: "foo",
+        version: "0.0.0",
+        dependencies: {},
+        devDependencies: {},
+      }),
+    );
+
+    {
+      const { err, out } = await command("install", pkg_name, "-y");
+      expect(err).toContain("Saved yarn.lock");
+      expect(out).toContain(`installed ${pkg_name}@${pkg_ver}`);
+    }
+
+    const yarn_lock_lines = (await file(join(package_dir, "yarn.lock")).text()).split(/\r?\n/g)
+    expect(yarn_lock_lines[0]).toBe("# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.");
+    expect(yarn_lock_lines[1]).toBe("# yarn lockfile v1");
+
+    const significant_lines = yarn_lock_lines.filter(s => s !== "" && s[0] !== "#" && s[s.length - 1] === ":" && s.includes(pkg_name))
```

Comment:
Can we also add a `expect(await file(join(package_dir, "yarn.lock")).text()).toMatchSnapshot()`?

Reply:
Done!